### PR TITLE
backport: merge bitcoin#27452, #29347, #29356, #29353, #29452, #29483, #30545, #31383 (BIP324 backports: part 5)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* When backporting [bitcoin#27452](https://github.com/bitcoin/bitcoin/pull/27452) in `feature_anchors.py`, `P2P_SERVICES` (`NODE_NETWORK | NODE_HEADERS_COMPRESSED`) has been replaced with `NODE_NETWORK` as the former evaluates to a value greater than `256` (specifically `2049`), which causes test failure. The replacement value is acceptable as `NODE_NETWORK` is the desired service flag expected by Dash Core ([source](https://github.com/dashpay/dash/blob/779e4295ad253bf909a7c45ec02743e580abc61b/src/protocol.cpp#L249-L254)).

  <details>

  <summary>Test failure:</summary>

  ```
  dash@89afd55ae77e:/src/dash$ ./test/functional/feature_anchors.py
  2024-12-14T12:31:22.244000Z TestFramework (INFO): PRNG seed is: 8365703189892653614
  2024-12-14T12:31:22.244000Z TestFramework (INFO): Initializing test directory /tmp/dash_func_test_j0ya02yu
  2024-12-14T12:31:22.776000Z TestFramework (INFO): When node starts, check if anchors.dat doesn't exist
  2024-12-14T12:31:22.776000Z TestFramework (INFO): Add 2 block-relay-only connections to node
  2024-12-14T12:31:24.781000Z TestFramework (INFO): Add 5 inbound connections to node
  2024-12-14T12:31:29.843000Z TestFramework (INFO): Check node connections
  2024-12-14T12:31:30.848000Z TestFramework (INFO): Check the addresses in anchors.dat
  2024-12-14T12:31:30.848000Z TestFramework (INFO): Perturb anchors.dat to test it doesn't throw an error during initialization
  2024-12-14T12:31:31.356000Z TestFramework (INFO): When node starts, check if anchors.dat doesn't exist anymore
  2024-12-14T12:31:31.357000Z TestFramework (INFO): Ensure addrv2 support
  2024-12-14T12:31:32.364000Z TestFramework (INFO): Add 256-bit-address block-relay-only connections to node
  2024-12-14T12:31:33.368000Z TestFramework (INFO): Check for addrv2 addresses in anchors.dat
  2024-12-14T12:31:33.369000Z TestFramework (ERROR): Unexpected exception caught during testing
  Traceback (most recent call last):
    File "/src/dash/test/functional/test_framework/test_framework.py", line 161, in main
      self.run_test()
    File "/src/dash/./test/functional/feature_anchors.py", line 135, in run_test
      new_data[services_index] = P2P_SERVICES
  ValueError: byte must be in range(0, 256)
  2024-12-14T12:31:33.870000Z TestFramework (INFO): Stopping nodes
  2024-12-14T12:31:33.871000Z TestFramework (WARNING): Not cleaning up dir /tmp/dash_func_test_j0ya02yu
  2024-12-14T12:31:33.871000Z TestFramework (ERROR): Test failed. Test logging available at /tmp/dash_func_test_j0ya02yu/test_framework.log
  ```

  </details>

## Breaking Changes

None expected.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK c6f23a718c81b7c574ba3fecbe52f342eda35a46

File: test/functional/feature_anchors.py [hunk 1]
```diff
@@ -7,11 +7,14 @@
 import os
 
 from test_framework.p2p import P2PInterface
```
Comment (PastaPastaPasta) [inline test/functional/feature_anchors.py:9]: why was P2P_SERVICES not added here?

**Review by PastaPastaPasta — APPROVED** [review]
utACK c6f23a718c81b7c574ba3fecbe52f342eda35a46;

looks good enough to me :) a few nits you can consider and if you want, can apply in another PR.

File: test/functional/feature_anchors.py (hunk 1 above)
Comment (kwvg) [reply test/functional/feature_anchors.py:9]: Please refer to pull request description for reasoning.

File: src/rpc/net.cpp [hunk 2]
```diff
@@ -360,7 +360,7 @@ static RPCHelpMan addconnection()
         {
             {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
             {"connection_type", RPCArg::Type::STR, RPCArg::Optional::NO, "Type of connection to open (\"outbound-full-relay\", \"block-relay-only\", \"addr-fetch\" or \"feeler\")."},
-            {"v2transport", RPCArg::Type::BOOL, RPCArg::Default{false}, "Attempt to connect using BIP324 v2 transport protocol"},
+            {"v2transport", RPCArg::Type::BOOL, RPCArg::Optional::NO, "Attempt to connect using BIP324 v2 transport protocol"},
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/rpc/net.cpp:363]: `self.Arg<bool>(2)` is the equivalent of `request.params[2].get_bool()`, the newer syntax being introduced in https://github.com/bitcoin/bitcoin/pull/28230 and fleshed out in https://github.com/bitcoin/bitcoin/pull/29277, backporting them out of order brings little benefit but would make backports until that point annoying at best, so it was decided to skip that change but not mark it as partial as the behaviour should be identical.

Though the `!request.params[2].isNull()` half of `use_v2transport` is a no-op as `v2transport` is no longer optional and is a nit that can be addressed later on.

**Pull request merged**